test/user_settings.h

```diff
@@ -29,24 +29,38 @@
 extern "C" {
 #endif
 
+/** Settings specific to the testing */
+/* #define WOLFHSM_CFG_TEST_VERBOSE */
+#define WOLFHSM_CFG_TEST_POSIX
+
+/** wolfHSM settings.  Simple overrides to show they work */
+#define WOLFHSM_CFG_COMM_DATA_LEN 1152
+
+#define WOLFHSM_CFG_NVM_OBJECT_COUNT 30
+#define WOLFHSM_CFG_SERVER_KEYCACHE_COUNT 10
+#define WOLFHSM_CFG_SERVER_KEYCACHE_SIZE 1024
+#define WOLFHSM_CFG_SERVER_DMAADDR_COUNT 8
+#define WOLFHSM_CFG_SERVER_CUSTOMCB_COUNT 6
```

Comment:
While this will work for our tests, I don't know if we want to establish the paradigm of putting wolfHSM configuration defines in `user_settings.h`. They should be separate. What if we don't want to use a wolfSSL user settings build? Then `WOLFSSL_USER_SETTINGS` won't be defined, and this header won't be included. 

I think we want a separate wolfHSM user settings file (how I had `wh_config.h`**) so it isn't tied to how wolfSSL is built. 

** or could rename to `wh_user_settings.h` if we want coherency with wolfSSL naming, but since our macros are `WOLFHSM_CFG` then a `wh_config.h` makes sense to me

Reply:
Yeah, was on the fence about this.  Really, we should have 2 different wolfHSM config files: wolfhsm_cfg_client.h and wolfhsm_cfg_server.h.  Since each side will/should have a separate user_settings, it made sense to steal some space in the same file to avoid any chances of misconfiguration.  

Lemme rework this into 2 files and pull out the CFG_TEST items since these are set by the Makefile.

Reply:
Corrected to wh_settings.h and wolfhsm_cfg.h.  Much cleaner solution.  Thanks for the guidance!